pkg/controller/shoot/shoot_maintenance_control.go

```diff
@@ -67,6 +69,7 @@ func (c *Controller) shootMaintenanceDelete(obj interface{}) {
 func (c *Controller) reconcileShootMaintenanceKey(key string) error {
 	namespace, name, err := cache.SplitMetaNamespaceKey(key)
 	if err != nil {
+		c.shootMaintenanceAdd(key)
```

Comment:
Why re-add the key here? The `err` probably rather indicates that there was an error with the key in general which means we probably shouldn't re-queue the key.

Reply:
Agreed, for this error case it doesn't make sense to re-add it as the error only occurs if the key itself is broken. Will fix.